packages/utils/test/formatingSpace.test.ts

```diff
@@ -0,0 +1,303 @@
+            const nodes = htmlToElements(`<p><a>
+     abc</a></p>`);
+            expect(removeFormattingSpace(nodes[0].childNodes[0].childNodes[0])).to.equal('abc');
+        });
+        it('(09) Should return trimmed text when multiples spaces before and after node', async () => {
+            const nodes = htmlToElements('<p>    <a>abc</a>    </p>');
+            expect(removeFormattingSpace(nodes[0].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(10) Should return trimmed text when multiple spaces after node', async () => {
+            const nodes = htmlToElements('<p><a>abc</a>   </p>');
+            expect(removeFormattingSpace(nodes[0].childNodes[0].childNodes[0])).to.equal('abc');
+        });
+        it('(11) Should return trimmed text when multiple spaces before node', async () => {
+            const nodes = htmlToElements('<p>   <a>abc</a></p>');
+            expect(removeFormattingSpace(nodes[0].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(12) Should return trimmed text when multiples spaces before and after node and text', async () => {
+            const nodes = htmlToElements('<p>    <a>  abc  </a>    </p>');
+            expect(removeFormattingSpace(nodes[0].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(13) Should return trimmed text when multiples spaces and line breaks before and after node and text', async () => {
+            const nodes = htmlToElements(`<p> 
+    <a>  abc  </a>  
+</p>`);
+            expect(removeFormattingSpace(nodes[0].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(14) Should return trimmed text when multiple spaces after both nodes (alt)', async () => {
+            const nodes = htmlToElements('<p><a>abc</a>   </p>   ');
+            expect(removeFormattingSpace(nodes[0].childNodes[0].childNodes[0])).to.equal('abc');
+        });
+        it('(15) Should return trimmed text when multiple spaces before both node (alt)', async () => {
+            const nodes = htmlToElements('   <p>   <a>abc</a></p>');
+            expect(removeFormattingSpace(nodes[1].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(16) Should return trimmed text when multiples spaces and line breaks before and after node and text (alt)', async () => {
+            const nodes = htmlToElements(`
+  <p> 
+    <a>  abc  </a>  
+  </p>
+`);
+            expect(removeFormattingSpace(nodes[1].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(17) should preserve a space before a line break at the end of a paragraph', async () => {
+            const nodes = htmlToElements('<p>abc <br><br></p>');
+            expect(removeFormattingSpace(nodes[0].childNodes[0])).to.equal('abc ');
+        });
+    });
+    describe('multiple childs nodes', () => {
+        // see : https://jsfiddle.net/5fzexc6L/9/ to have example of where white spaces should be kept between nodes
```

Comment:
Are you sure the link to jsfiddle is permanent?
Also: "... for examples clarifying where whitespace should be preserved."

Reply:
JSfiddle never expire normally : https://twitter.com/jsfiddle/status/137677807634481153